lib/spack/spack/test/concretize.py

```diff
@@ -525,3 +526,31 @@ def test_regression_issue_7941(self):
         t.concretize()
 
         assert s.dag_hash() == t.dag_hash()
+
+    @pytest.mark.parametrize('abstract_specs, checklist', [
```

Comment:
I spoke with @tgamblin and @scheibelp and we tried to think of a better way to express this test. How about
```
pytest.mark.parameterize(abstract_specs, [
                                           ['mpileaks', 'callpath@1.0', 'dyninst@8.1.1'],
                                           ['mpileaks', 'callpath@0.9', 'dyninst@8.1.1'],
                                           ['mpileaks', 'direct-mpich']])
def test_simultaneous_concretization_of_specs(self, abstract_specs):
    concrete_specs = spack.concretize.concretize_specs_together(*abstract_specs)
    names = set(dep.name for spec in concrete_specs for dep in spec.traverse())
    for name in names:
        name_specs = set(spec[name] for spec in concrete_specs if name in spec)
        assert len(name_specs) == 1
    for aspec in abstract_specs:
        assert any(aspec.satisfies(cspec) for cspec in concrete_specs)
```

Reply:
@alalazo I made another pass at reviewing this.